Index: xwiki-core/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java
===================================================================
--- xwiki-core/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java	(revision 19300)
+++ xwiki-core/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java	(working copy)
@@ -4421,7 +4421,9 @@
         if (objects.elementAt(bobj.getNumber()) == null) {
             return false;
         }
-        objects.set(bobj.getNumber(), null);
+
+        int pos = objects.indexOf(bobj);
+        objects.set(pos, null);
         addObjectsToRemove(bobj);
         return true;
     }
